packages/hardhat-core/src/internal/core/providers/construction.ts

```diff
@@ -69,6 +69,7 @@ export function createProvider(
       forkConfig = {
         jsonRpcUrl: hardhatNetConfig.forking?.url,
         blockNumber: hardhatNetConfig.forking?.blockNumber,
+        httpHeaders: hardhatNetConfig.forking.httpHeaders,
```

Comment:
Ugh, I should've made this into a separate commit, but @TimDaub this was the only thing that was missing from your PR, pretty close! :smile: